src/ansys/fluent/core/docker/docker_compose.py

```diff
@@ -0,0 +1,304 @@
+            ["docker", "--version"], capture_output=True, text=True, check=True
+        )
+        return True
+    except subprocess.CalledProcessError:
+        return False
+    except FileNotFoundError:
+        return False
+
+
+def _check_podman_installed():
+    """Check if Podman is installed."""
+    try:
+        result = subprocess.run(  # noqa: F841
+            ["podman", "--version"], capture_output=True, text=True, check=True
+        )
+        return True
+    except subprocess.CalledProcessError:
+        return False
+    except FileNotFoundError:
+        return False
+
+
+class ServerKey(str, enum.Enum):
+    """Keys for the servers launched through docker compose."""
+
+    MAIN = "main"
+
+
+@dataclasses.dataclass
+class DockerComposeLaunchConfig:
+    """Configuration options for launching Fluent through docker compose."""
+
+    image_name_fluent: str = dataclasses.field(
+        default="ghcr.io/ansys/pyfluent:latest",
+        metadata={METADATA_KEY_DOC: "Docker image running the Fluent gRPC server."},
+    )
+    keep_volume: bool = dataclasses.field(
+        default=False,
+        metadata={
+            METADATA_KEY_DOC: "If true, keep the volume after docker compose is stopped."
+        },
+    )
+    compose_file: str | None = dataclasses.field(
+        default=None,
+        metadata={
+            METADATA_KEY_DOC: (
+                "Docker compose file used to start the services. Uses the "
+                "'docker-compose.yaml' shipped with PyFluent by default."
+            ),
+            METADATA_KEY_NOPROMPT: True,
+        },
+    )
+
+
+def _set_env_vars(compose_name, container_dict):
+    """Generates environment variables for the Docker Compose setup.
+
+    Parameters
+    ----------
+    compose_name: str
+        The name of the compose instance.
+    container_dict: dict
+        A dictionary containing container configuration.
+    """
+
+    port = (
+        container_dict.get("fluent_port", "")
+        if container_dict.get("fluent_port")
+        else ""
+    )
+
+    ports = (
+        list(container_dict.get("ports").values())
+        if container_dict.get("ports")
+        else []
+    )
+
+    env_vars = {
+        "IMAGE_NAME": container_dict.get("fluent_image"),
+        "LICENSE_FILE": container_dict["environment"].get("ANSYSLMD_LICENSE_FILE"),
+        "REMOTE_PORTS": container_dict["environment"].get("REMOTING_PORTS"),
+        "NO_TRANSCRIPT": container_dict["environment"].get(
+            "FLUENT_NO_AUTOMATIC_TRANSCRIPT", "0"
+        ),
+        "COMMAND": " ".join(container_dict["command"]),
+        "MOUNT_TARGET": container_dict.get("mount_target"),
+        "MOUNT_SOURCE": container_dict.get("mount_source"),
+    }
+
+    env_vars["PORT_1"] = str(ports[0]) if ports else port
+    env_vars["PORT_2"] = str(ports[1]) if len(ports) > 1 else str(get_free_port())
+
+    for key, value in env_vars.items():
+        os.environ[key] = value
```

Comment:
Instead of setting these env vars in the host process, please send them through `subprocess.run`.

Reply:
@mkundu1 I have already tried this. 

It gives the following error.

![image](https://github.com/user-attachments/assets/dac2928e-8237-454b-bc1e-83a164d61ff8)

Reply:
@hpohekar Please check the `env` parameter in https://docs.python.org/3/library/subprocess.html#subprocess.run

Reply:
@mkundu1 We are setting them for this particular instance only. These env vars are erased once this instance gets closed so it does not affect other simultaneous instances, as we are already running multiple instances simultaneously in the GitHub and all of them are working fine without polluting original env vars of the system.

Reply:
> @hpohekar Please check the `env` parameter in https://docs.python.org/3/library/subprocess.html#subprocess.run

@mkundu1 I used the same `env` for both subprocess.run and subprocess.check_call, same error is there.

Reply:
Here is the exceprt from the standalone launcher code:
```
fluent_env = os.environ.copy()
fluent_env["var1"] = val1
fluent_env["var2"] = val2
...
subprocess.run(..., env=fluent_env)
```
What is the error you're seeing?

Reply:
@mkundu1 Done. Thank you. It's resolved after using `os.environ.copy()` then updating it with Fluent env vars. I was passing Fluent env vars dict directly.